prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -28,6 +28,7 @@ import java.io.ByteArrayInputStream
 import java.io.ByteArrayOutputStream
 import java.time.Instant
 import java.time.format.DateTimeFormatter
+import java.util.HashMap
```

Comment:
You shouldn't need java.util.HashMap in kotlin; use the built-in collection objs.

You'll need to refactor this code to follow kotlin style use of Map, instead of java style.  There are tons of examples in our code, or you can find good background info on various kotlin websites, or reach out to me or others for further help.